sim/core/utils.go

```diff
@@ -7,6 +7,22 @@ import (
 	"github.com/wowsims/tbc/sim/core/proto"
 )
 
+func MinInt(a int, b int) int {
```

Comment:
good news with next go version generics will mean you don't have to write a helper for every type anymore :D